src/main/java/org/sagebionetworks/bridge/spring/controllers/AuthenticationController.java

```diff
@@ -293,6 +299,59 @@ public StatusMessage resetPassword() throws Exception {
         return new StatusMessage("Password has been changed.");
     }
     
+    @PostMapping("/v3/auth/study")
+    public JsonNode changeStudy() throws Exception {
+        UserSession session = getAuthenticatedSession();
+        
+        // To switch studies, the account must be an administrative account with a Synapse User ID
+        StudyParticipant participant = session.getParticipant(); 
+        if (participant.getRoles().isEmpty()) {
+            throw new UnauthorizedException(STUDY_ACCESS_EXCEPTION_MSG);
+        } else if (participant.getSynapseUserId() == null) {
+            throw new BadRequestException("Account has not been assigned a Synapse user ID");
+        }
+        
+        // Retrieve the desired study
+        SignIn signIn = parseJson(SignIn.class);
+        String targetStudyId = signIn.getStudyId();
+        Study targetStudy = studyService.getStudy(targetStudyId);
+
+        // These are the accounts cross-study that are linked through the Synapse User ID
+        List<String> studyIds = accountDao.getStudyIdsForUser(participant.getSynapseUserId());
```

Comment:
Is it necessary to get all study IDs for the user when we already know which study ID they want? Would it be easier to just get the account for the specific study ID and and synapse user ID? (And a second call to getAccount for the api study in the ADMIN special case?)

Reply:
I've changed this to attempt loads of accounts (in API or the target study) to determine eligibility to switch studies.